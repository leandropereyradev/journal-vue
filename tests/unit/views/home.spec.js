import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home";
import mockRouter from "../mocks/mockRouter";

describe("Home View", () => {
  test("Should render the component correctly", () => {
    const wrapper = shallowMount(Home);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should redirect to no-entry on button click", () => {
    const wrapper = shallowMount(Home, {
      // global: opciones globales del componente
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "non-entry" });
  });
});
