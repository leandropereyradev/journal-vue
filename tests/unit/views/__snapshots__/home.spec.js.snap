// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Should render the component correctly 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
